Support proxy configs

Moving over from https://github.com/electron/electron/issues/7093

---

* Electron version:0.37.8
* Operating system:win7 32bit

spectron downloading failed 【getaddrinfo ENOTFOUND】
I have set the proxy in my PC as following。
proxy=http://username:password@proxy-url:8080/
https-proxy=http://username:password@proxy-url:8080/

---------------------------------------------------------------------------------------------
C:\tool\spectron-master\spectron-master>npm install

> electron-chromedriver@1.3.1 install C:\tool\spectron-master\spectron-master\no
de_modules\electron-chromedriver
> node ./download-chromedriver.js

**Downloading https://github.com/electron/electron/releases/download/v1.3.0/chrome
driver-v2.21-win32-ia32.zip failed: getaddrinfo ENOTFOUND github.com github.com:
443**
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\
node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v5.10.0
npm ERR! npm  v3.8.3
npm ERR! code ELIFECYCLE

npm ERR! electron-chromedriver@1.3.1 install: `node ./download-chromedriver.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the electron-chromedriver@1.3.1 install script 'node ./downlo
ad-chromedriver.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the electron-chromedriver
 package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./download-chromedriver.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs electron-chromedriver
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls electron-chromedriver
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\tool\spectron-master\spectron-master\npm-debug.log

--- a/download-chromedriver.js
+++ b/download-chromedriver.js
@@ -11,12 +11,18 @@ var baseUrl = process.env.NPM_CONFIG_ELECTRON_MIRROR ||
   process.env.electron_mirror ||
   'https://github.com/electron/electron/releases/download/v'
 
+var proxy = process.env.NPM_CONFIG_HTTPS_PROXY ||
+  process.env.npm_config_https_proxy ||
+  process.env.NPM_CONFIG_PROXY ||
+  process.env.npm_config_proxy
+
 var config = {
   baseUrl: baseUrl,
   // Sync minor version of package to minor version of Electron release
   electron: versionSegments[0] + '.' + versionSegments[1] + '.0',
   outputPath: path.join(__dirname, 'bin'),
-  version: 'v2.21'
+  version: 'v2.21',
+  proxy: proxy
 }
 
 function handleError (url, error) {
@@ -41,7 +47,7 @@ mkdirp(config.outputPath, function (error) {
 
   if (error) return handleError(fullUrl, error)
 
-  request.get({uri: fullUrl, encoding: null}, function (error, response, body) {
+  request.get({uri: fullUrl, encoding: null, proxy: config.proxy}, function (error, response, body) {
     if (error) return handleError(fullUrl, error)
     if (response.statusCode !== 200) return handleError(fullUrl, Error('Non-200 response (' + response.statusCode + ')'))
     unzip(body, function (error) {

